pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -586,6 +593,102 @@ func proxyConfigSamplingValue(config *meshconfig.ProxyConfig) float64 {
 	return sampling
 }
 
+func configureTracingProvider(hcmTracing *hcm.HttpConnectionManager_Tracing, pushCtx *model.PushContext, proxyCfg *meshconfig.ProxyConfig, node *model.Proxy) {
+	if proxyCfg == nil || proxyCfg.Tracing == nil {
+		return
+	}
+	var serviceCluster string
+	if node.XdsNode != nil {
+		serviceCluster = node.XdsNode.Cluster
+	}
+
+	switch tracer := proxyCfg.Tracing.Tracer.(type) {
+	case *meshconfig.Tracing_Datadog_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Datadog.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("datadog")
+			return
+		}
+
+		cfg, _ := datadogConfig(serviceCluster, hostname, clusterName)
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyDatadog,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_Lightstep_:
+		hostname, clusterName, err := getHostCluster(pushCtx, tracer.Lightstep.Address)
+		if err != nil {
+			model.IncLookupClusterFailures("lightstep")
+			return
+		}
+
+		cfg, _ := otelLightStepConfig(clusterName, hostname, tracer.Lightstep.GetAccessToken())
+		hcmTracing.Provider = &tracingcfg.Tracing_Http{
+			Name: envoyOpenTelemetry,
+			ConfigType: &tracingcfg.Tracing_Http_TypedConfig{
+				TypedConfig: cfg,
+			},
+		}
+	case *meshconfig.Tracing_OpenCensusAgent_:
+	case *meshconfig.Tracing_Stackdriver_:
+		// stackdriver/opencensus always need bootstrap tracing?
```

Comment:
De-facto, yes. OpenCensus used by these two can only be configured once, so even if you use Telemetry API and LDS, the settings are stuck until proxies restart.

Reply:
We do support it dynamically in the Telemetry API though I think (which is very high risk...).

In this case though, wouldn't the config be static through the proxy lifetime?